uppsrc/CtrlCore/X11DHCtrl.cpp

```diff
@@ -274,28 +274,28 @@ void DHCtrl::Terminate(void)
 
 	// Unmaps the window
 	MapWindow(false);
-
+	if(utop) {
 	// gathers data from XWindow (needs Input Context...)
-	XWindow *cw = XWindowFromWindow(top->window);
-
-	// Frees input context as needed
-	if(cw->xic)
-	{
-		XDestroyIC(cw->xic);
-		cw->xic = NULL;
+		XWindow *cw = XWindowFromWindow(utop->window);
+	
+		// Frees input context as needed
+		if(cw->xic)
```

Comment:
Shouldn't it be `if(cw && cw->xic)` { looks like risky situation for me...

Reply:
Done.